### PR TITLE
Allow array for fileMatch in yamlValidation contribution

### PR DESCRIPTION
**Pull request by JPinkney**
Signed-off-by: Josh Pinkney <joshpinkney@gmail.com>

### What does this PR do?
This PR is the client side part of https://github.com/redhat-developer/vscode-yaml/issues/387. Majority of it was taken from vscode-json-languageserver with minor modifications

### What issues does this PR fix or reference?
https://github.com/redhat-developer/vscode-yaml/issues/387

### Is it tested? How?

Add in:
```
"yamlValidation": [
      {
        "fileMatch": ["test.yml", "test.yaml"],
        "url": "https://gist.githubusercontent.com/JPinkney/fbbb95f3708e8b33f81267b243981ff6/raw/09af433aff3972f5d72cfe5c69951e94c1497875/kubernetes-console.json"
      }
]
```
under the "contributes" section of the package.json, then test the extension and make sure that the kubernetes schema is used for both test.yml and test.yaml (it might take a few seconds to load because the schema is so big)

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -10,16 +10,24 @@ import * as path from 'path';
...
+export interface ISchemaAssociation {
+  fileMatch: string[];
+  uri: string;
+}
+
 // eslint-disable-next-line @typescript-eslint/no-namespace
 namespace SchemaAssociationNotification {
-  export const type: NotificationType<ISchemaAssociations, unknown> = new NotificationType('json/schemaAssociations');
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  export const type: NotificationType<ISchemaAssociations | ISchemaAssociation[], any> = new NotificationType(
```
Comment (joshuawilson) [inline src/extension.ts:28]: Why is this an `any` here but the other PR uses `undefined`?

Comment (JPinkney) [reply src/extension.ts:28]: The yaml-language-server side notification has any as well:
```
export const type: NotificationType<ISchemaAssociations | SchemaConfiguration[], any> = new NotificationType(
    'json/schemaAssociations'
);
```

Comment (joshuawilson) [reply src/extension.ts:28]: OK. That is odd, I really I thought I read it as `undefined`.

**Review by evidolob — APPROVED** [review]
LGTM

File: src/extension.ts (hunk 1 above)
Comment (joshuawilson) [reply src/extension.ts:28]: I found it. 
https://github.com/redhat-developer/yaml-language-server/pull/348/files#diff-8a8ae07582c9d433ec8c2e5c4310ff8901e604f4965c5b90a49117ad46c47595R105

Comment (joshuawilson) [reply src/extension.ts:28]: not that matters, it is not used the same way

**Comment by joshuawilson** [comment]
If this is the PR that introduces the failure then it should be fixed here. Otherwise we block all other PRs.

**Comment by JPinkney** [comment]
It wasn't introduced here, it's present on master. The tests failed on https://github.com/redhat-developer/vscode-yaml/runs/1390809231?check_suite_focus=true with this PR https://github.com/redhat-developer/vscode-yaml/pull/390. I _think_ the newest vscode release makes it fail (came out Nov 6th) and thats why we see the error but didn't see the error on https://github.com/redhat-developer/vscode-yaml/pull/385

**Comment by joshuawilson** [comment]
Then merging this should be fine.

**Review by joshuawilson — APPROVED** [review]

**Pull request merged**